odoo/addons/base/models/ir_translation.py

```diff
@@ -425,9 +426,15 @@ def _sync_terms_translations(self, field, records):
                 else:
                     translations_to_match.append(translation)
 
+            terms_text_content = {}
+            if len(translations_to_match) > 0:
+                for term in terms:
+                    terms_text_content[lxml.html.fromstring(term).text_content()] = term
+
             for translation in translations_to_match:
-                matches = get_close_matches(translation.src, terms, 1, 0.9)
-                src = matches[0] if matches else None
+                src_text_content = lxml.html.fromstring(translation.src).text_content()
+                matches = get_close_matches(src_text_content, set(terms_text_content.keys()), 1, 0.9)
```

Comment:
Not good, as `set` makes this nondeterministic.  You can actually pass the dict itself in order to match against its keys.
```suggestion
                matches = get_close_matches(src_text_content, terms_text_content, 1, 0.9)
```

Reply:
done